src/Symfony/Component/Lock/Lock.php

```diff
@@ -32,22 +32,40 @@
     private $store;
     private $key;
     private $ttl;
+    private $autoRelease;
+    private $dirty = false;
 
     /**
-     * @param Key            $key
-     * @param StoreInterface $store
-     * @param float|null     $ttl
+     * @param Key            $key         Resource to lock
+     * @param StoreInterface $store       Store used to handle lock persistence
+     * @param float|null     $ttl         Maximum expected lock duration in seconds
+     * @param bool           $autoRelease Whether to automatically release the lock or not
      */
-    public function __construct(Key $key, StoreInterface $store, $ttl = null)
+    public function __construct(Key $key, StoreInterface $store, $ttl = null, $autoRelease = true)
     {
         $this->store = $store;
         $this->key = $key;
         $this->ttl = $ttl;
+        $this->autoRelease = (bool) $autoRelease;
 
         $this->logger = new NullLogger();
     }
 
     /**
+     * Automatically releases the underlying lock when the object is destructed.
+     */
+    public function __destruct()
+    {
+        if (!$this->autoRelease) {
```

Comment:
one "if" is enough in the method:
`if ($this->autoRelease && $this->dirty && $this->isAcquired()) { //...`